### PR TITLE
feat(amazonq): Implement file-level acceptance for /dev.

### PR DESCRIPTION
**Pull request by ctidd**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Provides the feature for a user to accept individual generated files, and add them to their project.

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [x] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by neilk-aws — APPROVED** [review]

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt [hunk 1]
```diff
@@ -244,21 +267,84 @@ class FeatureDevController(
...
+                } else {
+                    pollAttempt++
+                    delay(pollDelayMs)
+                }
+            }
+        }
+
+        val nextInsertAction = insertAction()
+
+        println("nextInsertAction: $filePaths, $deletedFiles")
```
Comment (manodnyab) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:339]: can be deleted?

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt [hunk 2]
```diff
@@ -325,17 +412,22 @@ class FeatureDevController(
                 }
             }
 
+            val rejectedFilesCount = filePaths.count { it.rejected } + deletedFiles.count { it.rejected }
+            val acceptedFilesCount = filePaths.count { it.changeApplied } + filePaths.count { it.changeApplied }
+            val remainingFilesCount = filePaths.count() + deletedFiles.count() - acceptedFilesCount - rejectedFilesCount
+
             AmazonqTelemetry.isAcceptedCodeChanges(
-                amazonqNumberOfFilesAccepted = (filePaths.filterNot { it.rejected }.size + deletedFiles.filterNot { it.rejected }.size) * 1.0,
+                amazonqNumberOfFilesAccepted = remainingFilesCount * 1.0,
```
Comment (manodnyab) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:420]: why are we multiplying this by 1.0?

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/Session.kt [hunk 3]
```diff
@@ -82,24 +85,65 @@ class Session(val tabID: String, val project: Project) {
...
+        deletedFiles: List<DeletedFileInfo>,
+        messenger: MessagePublisher,
+    ) {
+        if (this._codeResultMessageId.isNullOrEmpty()) {
+            return
+        }
+
+        val codeResultMessageId = this._codeResultMessageId
+        if (codeResultMessageId != null) {
+            messenger.updateFileComponent(this.tabID, filePaths, deletedFiles, codeResultMessageId, true)
```
Comment (manodnyab) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/Session.kt:144]: Could you please add the parameter name before the boolean value?

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt (hunk 2 above)
Comment (ctidd) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:420]: Coercing to double; updating to `.toDouble()` to be more explicit.

**Review by rli — APPROVED** [review]

**Pull request merged**